python/tests/test_highlevel.py

```diff
@@ -1985,6 +1985,30 @@ def test_sites(self):
             assert mutation_index == len(mutations)
         assert some_sites
 
+    def test_site_alleles(self):
```

Comment:
No real point in running a simulatino here and smaller tests would be better. Could do something like

```
class TestSiteAlleles:
     def test_no_mutations(self):
          tables = tskit.TableCollection()
          tables.sites.add_row(0, ancestral_state="A") 
          site = tables.tree_sequence().site(0)
          assert site.alleles == {"A"}

     @pytest.mark.parametrize(k, range(5))
     def test_k_mutations(self):
          tables = tskit.TableCollection()
          tables.sites.add_row(0, ancestral_state="-1") 
          for j in range(k):
                tables.mutations.add_row(site=0, node=0, derived_state=str(j))
           assert stuff
```